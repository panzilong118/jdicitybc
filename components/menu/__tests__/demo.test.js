import * as React from "react";

import renderer from 'react-test-renderer';

import Horizontal from '../demo/Horizontal';

import Inline from '../demo/Inline';

import InlineCollapsed from '../demo/InlineCollapsed';

import SiderCurrent from '../demo/SiderCurrent';

// import SwitchMode from '../demo/SwitchMode';

// import Theme from '../demo/Theme';

import Vertical from '../demo/Vertical';


describe('<Menu />', () => {
    
    it('Horizontal top navigation menu.', () => {
        const demo = renderer.create(<Horizontal />);
        let snapsDemo = demo.toJSON();
        expect(snapsDemo).toMatchSnapshot();
    });

    it('Horizontal top navigation menu.', () => {
        const demo = renderer.create(<Inline />);
        let snapsDemo = demo.toJSON();
        expect(snapsDemo).toMatchSnapshot();
    });

    it("Inline menu could be collapsed", ()=>{
        const demo = renderer.create(<InlineCollapsed />);
        let snapsDemo = demo.toJSON();
        expect(snapsDemo).toMatchSnapshot();
    });

    it("点击菜单，收起其他展开的所有菜单，保持菜单聚焦简洁", ()=>{
        const demo = renderer.create(<SiderCurrent />);
        let snapsDemo = demo.toJSON();
        expect(snapsDemo).toMatchSnapshot();
    });

    // it("Show the dynamic switching mode (between 'inline' and 'vertical').", ()=>{
    //     const demo = renderer.create(<SwitchMode />);
    //     let snapsDemo = demo.toJSON();
    //     expect(snapsDemo).toMatchSnapshot();
    // });

    // it("There are two built-in themes: 'light' and 'dark'. The default value is 'light').", ()=>{
    //     const demo = renderer.create(<Theme />);
    //     let snapsDemo = demo.toJSON();
    //     expect(snapsDemo).toMatchSnapshot();
    // });
    
    it("Submenus open as pop-ups", ()=>{
        const demo = renderer.create(<Vertical />);
        let snapsDemo = demo.toJSON();
        expect(snapsDemo).toMatchSnapshot();
    });
    
});
