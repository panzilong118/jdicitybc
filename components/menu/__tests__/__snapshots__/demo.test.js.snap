// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> Horizontal top navigation menu. 1`] = `
<ul
  aria-activedescendant=""
  className="jc-menu jc-menu-light jc-menu-root jc-menu-horizontal"
  onKeyDown={[Function]}
  role="menu"
  style={Object {}}
  tabIndex="0"
>
  <li
    aria-disabled={undefined}
    aria-selected={true}
    className="jc-menu-item jc-menu-item-selected"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    role="menuitem"
    style={Object {}}
    title={undefined}
  >
    <i
      className="jcicon jcicon-mail "
      style={Object {}}
    />
    Navigation One
  </li>
  <li
    aria-disabled={true}
    aria-selected={false}
    className="jc-menu-item jc-menu-item-disabled"
    role="menuitem"
    style={Object {}}
    title={undefined}
  >
    <i
      className="jcicon jcicon-appstore "
      style={Object {}}
    />
    Navigation Two
  </li>
  <li
    className="jc-menu-submenu jc-menu-submenu-horizontal"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={undefined}
  >
    <div
      aria-expanded={false}
      aria-haspopup="true"
      aria-owns="item_2$Menu"
      className="jc-menu-submenu-title"
      onBlur={undefined}
      onClick={[Function]}
      onContextMenu={undefined}
      onFocus={undefined}
      onMouseDown={undefined}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={undefined}
      style={Object {}}
      title={undefined}
    >
      <span>
        <i
          className="jcicon jcicon-setting "
          style={Object {}}
        />
        Navigation Three - Submenu
      </span>
      <i
        className="jc-menu-submenu-arrow"
      />
    </div>
  </li>
  <li
    aria-disabled={undefined}
    aria-selected={false}
    className="jc-menu-item"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    role="menuitem"
    style={Object {}}
    title={undefined}
  >
    <a
      href="https://ant.design"
      rel="noopener noreferrer"
      target="_blank"
    >
      Navigation Four - Link
    </a>
  </li>
</ul>
`;

exports[`<Menu /> Horizontal top navigation menu. 2`] = `
<ul
  aria-activedescendant=""
  className="jc-menu jc-menu-light jc-menu-root jc-menu-inline"
  onKeyDown={[Function]}
  role="menu"
  style={
    Object {
      "width": 256,
    }
  }
  tabIndex="0"
>
  <li
    className="jc-menu-submenu jc-menu-submenu-inline jc-menu-submenu-open jc-menu-submenu-selected"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={undefined}
  >
    <div
      aria-expanded={true}
      aria-haspopup="true"
      aria-owns="sub1$Menu"
      className="jc-menu-submenu-title"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "paddingLeft": 24,
        }
      }
      title={undefined}
    >
      <span>
        <i
          className="jcicon jcicon-mail "
          style={Object {}}
        />
        <span>
          Navigation One
        </span>
      </span>
      <i
        className="jc-menu-submenu-arrow"
      />
    </div>
    <ul
      aria-activedescendant=""
      className="jc-menu  jc-menu-sub jc-menu-inline"
      id="sub1$Menu"
      role="menu"
      style={Object {}}
    >
      <li
        className=" jc-menu-item-group"
      >
        <div
          className="jc-menu-item-group-title"
          title="Item 1"
        >
          Item 1
        </div>
        <ul
          className="jc-menu-item-group-list"
        >
          <li
            aria-disabled={undefined}
            aria-selected={true}
            className="jc-menu-item jc-menu-item-selected"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            role="menuitem"
            style={
              Object {
                "paddingLeft": 48,
              }
            }
            title={undefined}
          >
            Option 1
          </li>
          <li
            aria-disabled={undefined}
            aria-selected={false}
            className="jc-menu-item"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            role="menuitem"
            style={
              Object {
                "paddingLeft": 48,
              }
            }
            title={undefined}
          >
            Option 2
          </li>
        </ul>
      </li>
      <li
        className=" jc-menu-item-group"
      >
        <div
          className="jc-menu-item-group-title"
          title="Item 2"
        >
          Item 2
        </div>
        <ul
          className="jc-menu-item-group-list"
        >
          <li
            aria-disabled={undefined}
            aria-selected={false}
            className="jc-menu-item"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            role="menuitem"
            style={
              Object {
                "paddingLeft": 48,
              }
            }
            title={undefined}
          >
            Option 3
          </li>
          <li
            aria-disabled={undefined}
            aria-selected={false}
            className="jc-menu-item"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            role="menuitem"
            style={
              Object {
                "paddingLeft": 48,
              }
            }
            title={undefined}
          >
            Option 4
          </li>
        </ul>
      </li>
    </ul>
  </li>
  <li
    className="jc-menu-submenu jc-menu-submenu-inline"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={undefined}
  >
    <div
      aria-expanded={false}
      aria-haspopup="true"
      aria-owns="sub2$Menu"
      className="jc-menu-submenu-title"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "paddingLeft": 24,
        }
      }
      title={undefined}
    >
      <span>
        <i
          className="jcicon jcicon-appstore "
          style={Object {}}
        />
        <span>
          Navigation Two
        </span>
      </span>
      <i
        className="jc-menu-submenu-arrow"
      />
    </div>
  </li>
  <li
    className="jc-menu-submenu jc-menu-submenu-inline"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={undefined}
  >
    <div
      aria-expanded={false}
      aria-haspopup="true"
      aria-owns="sub4$Menu"
      className="jc-menu-submenu-title"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "paddingLeft": 24,
        }
      }
      title={undefined}
    >
      <span>
        <i
          className="jcicon jcicon-setting "
          style={Object {}}
        />
        <span>
          Navigation Three
        </span>
      </span>
      <i
        className="jc-menu-submenu-arrow"
      />
    </div>
  </li>
</ul>
`;

exports[`<Menu /> Inline menu could be collapsed 1`] = `
<div
  style={
    Object {
      "width": 256,
    }
  }
>
  <button
    className="jc-btn jc-btn-primary"
    onClick={[Function]}
    style={
      Object {
        "marginBottom": 16,
      }
    }
    type="button"
  >
    <i
      className="jcicon jcicon-menu-fold "
      style={Object {}}
    />
  </button>
  <ul
    aria-activedescendant=""
    className="jc-menu jc-menu-dark jc-menu-root jc-menu-inline"
    onKeyDown={[Function]}
    role="menu"
    style={Object {}}
    tabIndex="0"
  >
    <li
      aria-disabled={undefined}
      aria-selected={true}
      className="jc-menu-item jc-menu-item-selected"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="menuitem"
      style={
        Object {
          "paddingLeft": 24,
        }
      }
      title={undefined}
    >
      <i
        className="jcicon jcicon-pie-chart "
        style={Object {}}
      />
      <span>
        Option 1
      </span>
    </li>
    <li
      aria-disabled={undefined}
      aria-selected={false}
      className="jc-menu-item"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="menuitem"
      style={
        Object {
          "paddingLeft": 24,
        }
      }
      title={undefined}
    >
      <i
        className="jcicon jcicon-desktop "
        style={Object {}}
      />
      <span>
        Option 2
      </span>
    </li>
    <li
      aria-disabled={undefined}
      aria-selected={false}
      className="jc-menu-item"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="menuitem"
      style={
        Object {
          "paddingLeft": 24,
        }
      }
      title={undefined}
    >
      <i
        className="jcicon jcicon-inbox "
        style={Object {}}
      />
      <span>
        Option 3
      </span>
    </li>
    <li
      className="jc-menu-submenu jc-menu-submenu-inline jc-menu-submenu-open"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={undefined}
    >
      <div
        aria-expanded={true}
        aria-haspopup="true"
        aria-owns="sub1$Menu"
        className="jc-menu-submenu-title"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "paddingLeft": 24,
          }
        }
        title={undefined}
      >
        <span>
          <i
            className="jcicon jcicon-mail "
            style={Object {}}
          />
          <span>
            Navigation One
          </span>
        </span>
        <i
          className="jc-menu-submenu-arrow"
        />
      </div>
      <ul
        aria-activedescendant=""
        className="jc-menu  jc-menu-sub jc-menu-inline"
        id="sub1$Menu"
        role="menu"
        style={Object {}}
      >
        <li
          aria-disabled={undefined}
          aria-selected={false}
          className="jc-menu-item"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="menuitem"
          style={
            Object {
              "paddingLeft": 48,
            }
          }
          title={undefined}
        >
          Option 5
        </li>
        <li
          aria-disabled={undefined}
          aria-selected={false}
          className="jc-menu-item"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="menuitem"
          style={
            Object {
              "paddingLeft": 48,
            }
          }
          title={undefined}
        >
          Option 6
        </li>
        <li
          aria-disabled={undefined}
          aria-selected={false}
          className="jc-menu-item"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="menuitem"
          style={
            Object {
              "paddingLeft": 48,
            }
          }
          title={undefined}
        >
          Option 7
        </li>
        <li
          aria-disabled={undefined}
          aria-selected={false}
          className="jc-menu-item"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="menuitem"
          style={
            Object {
              "paddingLeft": 48,
            }
          }
          title={undefined}
        >
          Option 8
        </li>
      </ul>
    </li>
    <li
      className="jc-menu-submenu jc-menu-submenu-inline"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={undefined}
    >
      <div
        aria-expanded={false}
        aria-haspopup="true"
        aria-owns="sub2$Menu"
        className="jc-menu-submenu-title"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "paddingLeft": 24,
          }
        }
        title={undefined}
      >
        <span>
          <i
            className="jcicon jcicon-appstore "
            style={Object {}}
          />
          <span>
            Navigation Two
          </span>
        </span>
        <i
          className="jc-menu-submenu-arrow"
        />
      </div>
    </li>
  </ul>
</div>
`;

exports[`<Menu /> Submenus open as pop-ups 1`] = `
<ul
  aria-activedescendant=""
  className="jc-menu jc-menu-light jc-menu-root jc-menu-vertical"
  onKeyDown={[Function]}
  role="menu"
  style={
    Object {
      "width": 256,
    }
  }
  tabIndex="0"
>
  <li
    className="jc-menu-submenu jc-menu-submenu-vertical"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={undefined}
  >
    <div
      aria-expanded={false}
      aria-haspopup="true"
      aria-owns="sub1$Menu"
      className="jc-menu-submenu-title"
      onBlur={undefined}
      onClick={[Function]}
      onContextMenu={undefined}
      onFocus={undefined}
      onMouseDown={undefined}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={undefined}
      style={Object {}}
      title={undefined}
    >
      <span>
        <i
          className="jcicon jcicon-mail "
          style={Object {}}
        />
        <span>
          Navigation One
        </span>
      </span>
      <i
        className="jc-menu-submenu-arrow"
      />
    </div>
  </li>
  <li
    className="jc-menu-submenu jc-menu-submenu-vertical"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={undefined}
  >
    <div
      aria-expanded={false}
      aria-haspopup="true"
      aria-owns="sub2$Menu"
      className="jc-menu-submenu-title"
      onBlur={undefined}
      onClick={[Function]}
      onContextMenu={undefined}
      onFocus={undefined}
      onMouseDown={undefined}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={undefined}
      style={Object {}}
      title={undefined}
    >
      <span>
        <i
          className="jcicon jcicon-appstore "
          style={Object {}}
        />
        <span>
          Navigation Two
        </span>
      </span>
      <i
        className="jc-menu-submenu-arrow"
      />
    </div>
  </li>
  <li
    className="jc-menu-submenu jc-menu-submenu-vertical"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={undefined}
  >
    <div
      aria-expanded={false}
      aria-haspopup="true"
      aria-owns="sub4$Menu"
      className="jc-menu-submenu-title"
      onBlur={undefined}
      onClick={[Function]}
      onContextMenu={undefined}
      onFocus={undefined}
      onMouseDown={undefined}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={undefined}
      style={Object {}}
      title={undefined}
    >
      <span>
        <i
          className="jcicon jcicon-setting "
          style={Object {}}
        />
        <span>
          Navigation Three
        </span>
      </span>
      <i
        className="jc-menu-submenu-arrow"
      />
    </div>
  </li>
</ul>
`;

exports[`<Menu /> 点击菜单，收起其他展开的所有菜单，保持菜单聚焦简洁 1`] = `
<ul
  aria-activedescendant=""
  className="jc-menu jc-menu-light jc-menu-root jc-menu-inline"
  onKeyDown={[Function]}
  role="menu"
  style={
    Object {
      "width": 256,
    }
  }
  tabIndex="0"
>
  <li
    className="jc-menu-submenu jc-menu-submenu-inline jc-menu-submenu-open"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={undefined}
  >
    <div
      aria-expanded={true}
      aria-haspopup="true"
      aria-owns="sub1$Menu"
      className="jc-menu-submenu-title"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "paddingLeft": 24,
        }
      }
      title={undefined}
    >
      <span>
        <i
          className="jcicon jcicon-mail "
          style={Object {}}
        />
        <span>
          Navigation One
        </span>
      </span>
      <i
        className="jc-menu-submenu-arrow"
      />
    </div>
    <ul
      aria-activedescendant=""
      className="jc-menu  jc-menu-sub jc-menu-inline"
      id="sub1$Menu"
      role="menu"
      style={Object {}}
    >
      <li
        aria-disabled={undefined}
        aria-selected={false}
        className="jc-menu-item"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        role="menuitem"
        style={
          Object {
            "paddingLeft": 48,
          }
        }
        title={undefined}
      >
        Option 1
      </li>
      <li
        aria-disabled={undefined}
        aria-selected={false}
        className="jc-menu-item"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        role="menuitem"
        style={
          Object {
            "paddingLeft": 48,
          }
        }
        title={undefined}
      >
        Option 2
      </li>
      <li
        aria-disabled={undefined}
        aria-selected={false}
        className="jc-menu-item"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        role="menuitem"
        style={
          Object {
            "paddingLeft": 48,
          }
        }
        title={undefined}
      >
        Option 3
      </li>
      <li
        aria-disabled={undefined}
        aria-selected={false}
        className="jc-menu-item"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        role="menuitem"
        style={
          Object {
            "paddingLeft": 48,
          }
        }
        title={undefined}
      >
        Option 4
      </li>
    </ul>
  </li>
  <li
    className="jc-menu-submenu jc-menu-submenu-inline"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={undefined}
  >
    <div
      aria-expanded={false}
      aria-haspopup="true"
      aria-owns="sub2$Menu"
      className="jc-menu-submenu-title"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "paddingLeft": 24,
        }
      }
      title={undefined}
    >
      <span>
        <i
          className="jcicon jcicon-appstore "
          style={Object {}}
        />
        <span>
          Navigation Two
        </span>
      </span>
      <i
        className="jc-menu-submenu-arrow"
      />
    </div>
  </li>
  <li
    className="jc-menu-submenu jc-menu-submenu-inline"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={undefined}
  >
    <div
      aria-expanded={false}
      aria-haspopup="true"
      aria-owns="sub4$Menu"
      className="jc-menu-submenu-title"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      style={
        Object {
          "paddingLeft": 24,
        }
      }
      title={undefined}
    >
      <span>
        <i
          className="jcicon jcicon-setting "
          style={Object {}}
        />
        <span>
          Navigation Three
        </span>
      </span>
      <i
        className="jc-menu-submenu-arrow"
      />
    </div>
  </li>
</ul>
`;
